test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1723,7 +1720,9 @@ private synchronized void stopNodesAndClients(Collection<NodeAndClient> nodeAndC
 
         for (NodeAndClient nodeAndClient: nodeAndClients) {
             removeDisruptionSchemeFromNode(nodeAndClient);
-            NodeAndClient previous = nodes.remove(nodeAndClient.name);
+            final NavigableMap<String, NodeAndClient> newNodes = new TreeMap<>(nodes);
```

Comment:
This is duplicated at least once, I think we should refactor it into a method.

Reply:
done :)